src/evaluators/utility/PHAL_SharedParameter.hpp

```diff
@@ -8,27 +8,38 @@
 #include "Phalanx_Evaluator_WithBaseImpl.hpp"
 #include "Phalanx_Evaluator_Derived.hpp"
 #include "Sacado_ParameterAccessor.hpp"
+#include "Albany_Accessors.hpp"
 
 //IKT 6/3/2020 TODO: implement support for vector parameters, which is not available currently.
 
 namespace PHAL
 {
 
-template<typename EvalT, typename Traits, typename ParamNameEnum, ParamNameEnum ParamName>
+template<typename EvalT, typename Traits, typename ParamNameEnum>
 class SharedParameter : public PHX::EvaluatorWithBaseImpl<Traits>,
                         public PHX::EvaluatorDerived<EvalT, Traits>,
                         public Sacado::ParameterAccessor<EvalT, SPL_Traits>
 {
+private:
+  std::shared_ptr<Albany::Accessor<typename EvalT::ScalarT>> accessor;
+
 public:
 
   typedef typename EvalT::ScalarT   ScalarT;
   typedef ParamNameEnum             EnumType;
 
-  SharedParameter (const Teuchos::ParameterList& p, const Teuchos::RCP<Albany::Layouts>& dl)
+  SharedParameter (Teuchos::ParameterList& p, const Teuchos::RCP<Albany::Layouts>& dl)
   {
     param_name   = p.get<std::string>("Parameter Name");
     param_as_field = PHX::MDField<ScalarT,Dim>(param_name,dl->shared_param);
 
+    Teuchos::RCP<Albany::Accessors<EvalT>> accessors = p.get<Teuchos::RCP<Albany::AccessorsMap>>("Accessors")->at<EvalT>();
+    if (accessors->accessors.count(param_name)==0) {
```

Comment:
I think it shouldn't be SharedParameter creating the accessors...It should be the problem class. Then, you could actually already pass the right accessor (for the right ScalarT) to the ctor of SharedParam, rather than pass the whole map.